proposals/extend_pod_security_admission.md

```diff
@@ -0,0 +1,229 @@
+
+Here is an example of this request: [Grant specific permissions for specific services](https://github.com/kubernetes/pod-security-admission/issues/1).
+
+A solution would be to use microsegmentation, and isolate privileged pods to their own namespace. However, this introduces other complexities such as managing networking across pods across namespaces, managing permissions, etc.
+
+There are a few other considerations with PSA:
+
+* For users, violations are visible only for `warn` and `enforce` mode. With `audit` there's no trace that users can track unless the audit trace is enabled via the API-server.
+* The violations are only returned along with admission responses. And the warning messages are not easy to parse. It exposes a metrics endpoint to record [evaluations](https://github.com/kubernetes/pod-security-admission/blob/5219c1944103298680f1298e30155ce541af8a02/metrics/metrics.go#L43-L47).
+* No mutation ability.
+* No support for any other resources than Pods.
+* Workload controllers (ex., Deployments) [are not blocked](https://kubernetes.io/docs/concepts/security/pod-security-admission/#workload-resources-and-pod-templates) in `enforce` mode, only the Pods created indirectly by them.
+
+## Proposed Solution
+
+PSA is an in-tree solution for pod security compliance, and it seems best to enable users to leverage the built-in solution for pod security enforcement. Kyverno can be integrated with PSA to extend its ability and provide fine grained checks.
+
+With this solution, Kyverno would reuse the in-tree pod security implementation and not require users to manage multiple Kyverno policies, one for each control.  
+
+### Extend PSA via a new rule
+
+Kyverno can extend PSS profiles with a new validation rule, `validate.podSecurity`.
+
+The rule will contain a "profile" that is set to "restricted" by default, and a list of exclusions that the user wants to allow. When set to "restricted", Kyverno will apply both Baseline and Restricted pod security profiles and block on violations. 
+
+If the PSA is enabled with "enforce=restricted", there's not much left for this Kyverno rule to do. Only if the PSA is in "enforce=baseline" or "enforce=privileged" mode, users can use the Kyverno rule for granular control.
+
+The `podSecurity` validate rule applies configured [pod security profile level](https://kubernetes.io/docs/concepts/security/pod-security-admission/#pod-security-levels) to the selected namespaces. You can define exemptions from enforcement when creating pods. Exemptions can be configured via the `podSecurity.exclude` attribute.
+
+For example, if a user wants to apply Restricted PSS to the selected namespaces "test" and "staging" but to skip checking the Control "Capabilities" for pods running "ghcr.io/example/nginx:1.2.3", the following policy does the job.
+
+```yaml
+validationFailureAction: enforce
+rules:
+  - name: validate-host-network
+    match:
```

Comment:
This is a nit, but can we make sure that all policies referenced herein use `any`|`all` under the `match`|`exclude` blocks so they're consistent with current guidance?

Reply:
Arf... always forgot this. Updated.